### PR TITLE
Improved KeyCloak usage

### PR DESCRIPTION
**Pull request by inbarbarkai**
1. Implemented decorator for validating user role.
2. Added additional seed to KeyCloak

### CODE REVIEW

File: init/keycloak/kc-seed.sh [hunk 1]
```diff
@@ -7,5 +7,26 @@ do
...
+./kcadm.sh create clients \
+    -s clientId=invitease \
+    -s 'redirectUris=["http://invitease.localhost/*"]' \
+    -s clientAuthenticatorType=client-secret -s serviceAccountsEnabled=true \
+    -s secret=d0b8122f-8dfb-46b7-b68a-f5cc4e25d000 
+CLIENT_ID=$(./kcadm.sh get clients -r master -q clientId=invitease --fields id | grep -oP '(?<="id" : ")[^"]*')
+./kcadm.sh create groups -r master -s name=admin
+GROUP_ID=$(./kcadm.sh get groups -r master -q name=admin | grep -oP '(?<="id" : ")[^"]*')
+./kcadm.sh update users/$USER_ID/groups/$GROUP_ID -r master -s realm=master -s userId=$USER_ID -s groupId=$GROUP_ID -n
+./kcadm.sh create clients/$CLIENT_ID/protocol-mappers/models -r master \
```
Comment (yarons) [inline init/keycloak/kc-seed.sh:20]: What do you think about adding phone, user_role (admin, guard, user) and NPP/NPR field already as part of this PR?

**Comment by yarons** [comment]
Just a general question, not sure it's relevant to this PR, do you think we should use the dedicated fastapi-keycloak python lib?
https://fastapi-keycloak.code-specialist.com/

Thanks :)

File: init/keycloak/kc-seed.sh (hunk 1 above)
Comment (inbarbarkai) [reply init/keycloak/kc-seed.sh:20]: I'm not sure what you mean

**Comment by inbarbarkai** [comment]
Not relevant for this PR but I'll definitely take a look

File: init/keycloak/kc-seed.sh (hunk 1 above)
Comment (inbarbarkai) [reply init/keycloak/kc-seed.sh:20]: Not sure what you mean

Comment (yarons) [reply init/keycloak/kc-seed.sh:20]: Here we're defining the user fields, am I correct?

Comment (inbarbarkai) [reply init/keycloak/kc-seed.sh:20]: No, I just create a group and add the admin user to it

**Review by yarons — APPROVED** [review]
Amazing!!!

**Comment by yarons** [comment]
Squash it up!

**Comment by inbarbarkai** [comment]
> Just a general question, not sure it's relevant to this PR, do you think we should use the dedicated fastapi-keycloak python lib? https://fastapi-keycloak.code-specialist.com/
> 
> Thanks :)

This requires the admin cli secret which I don't want to expose

**Pull request merged**